frontend/src/components/CreateModal.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { Component } from 'react';
+import { Modal, Input, InputNumber, Button, Checkbox, Table, Tag } from 'antd';
+import _ from 'lodash';
+import axios from 'axios';
+
+import { api_endpoints } from '../helpers/constants'
+
+
+class CreateModal extends Component {
+  state = {
+    office_name:'',
+    street: '',
+    city: '',
+    postal_code: '',
+    monthly_rent: '',
+    company_name: '',
+    headquarter: false,
+    offices: []
+  }
+
+  handleAdd = () => {
+    const { office_name, street, city, postal_code, monthly_rent, headquarter } = this.state
+    this.setState({
+      offices: [...this.state.offices, {
+        key: (this.state.offices.length + 1).toString(),
+        name: office_name,
+        street,
+        city,
+        postal_code,
+        monthly_rent,
+        headquarter
+      }]
+    },()=>{
+      console.log(this.state.offices)
+
+      this.setState({
+        office_name: '',
+        street: '',
+        city: '',
+        postal_code: '',
+        monthly_rent: '',
+        headquarter: false
+      })
+    })
+  }
+
+  columns = [
+    {
+      title: 'Office Name',
+      dataIndex: 'name',
+      key: 'name',
+      render: text => <span>{text}</span>
+    },
+    {
+      title: 'Street',
+      dataIndex: 'street',
+      key: 'street',
+      render: text => <span>{text}</span>
+    },
+    {
+      title: 'Other',
+      dataIndex: 'tag',
+      key: 'headquarter',
+      render: (value, record) => {
+        if (record.headquarter) {
+          return (
+            <Tag color="green" >
+              Headquarter
+            </Tag>
+          )
+        }
+
+      },
+    }
+  ]
+
+  handleCancel = () => {
+    this.setState({
+      offices: []
+    }, () => {
+      this.props.onCancel()
+    })
+  }
+
+  handleOk = async () => {
+    const { company_name, offices } = this.state
+    const { onSave } = this.props
+    await axios.post(api_endpoints.CREATE_COMPANY_OFFICE, {
+      name: company_name,
+      offices
+    })
+    onSave()
+  }
+
+  render() {
+    const { ...props } = this.props
+    const { office_name, city, street, postal_code, monthly_rent, headquarter, offices } = this.state
+    return (
+      <Modal
+        {...props}
+        onCancel={() => this.handleCancel()}
+        onOk={() => this.handleOk()}
+      >
+        <p>
+          <b>Company Name:</b>  <br/>
+          <Input placeholder='Company Name' onChange={({ target: { value: company_name } }) => this.setState({ company_name })} />
```

Comment:
Here, we can use more generic solution for handling onChange.
like 
onChange = ({ target: { value, id }) => {
    this.setState({ id: value })
}
`<Input placeholder='Company Name' onChange={this.onChange()} id='company_name' />`

Reply:
Very interesting approach. 
```
onChange = ({target: {value, id}}) => {
    this.setState({[id]: value})
}
```
It should work. Thanks for your good approach.